### PR TITLE
Added tests folder and made minor changes to utils.R

### PR DESCRIPTION
**Pull request by itismeghasyam**

### CODE REVIEW

**Comment by th1vairam** [comment]
Why there is a utils.R in test folder? can you remove that and recommit. Other that everything looks great.

**Comment by philerooski** [comment]
Tests (test_* files) need to live in the tests/testthat directory. 

So the directory structure looks like:

tests
+-- data
│   +-- hr_test.json
│   \\-- phone_data_test.json
+-- testthat
│   +-- test_getHR.R
│   +-- test_getTremorFeatures.R
│   +-- test_getWalkFeatures.R
│   +-- test_sensors.R
│   \\-- test_utils.R
\\-- testthat.R

**Comment by itismeghasyam** [comment]
Done.

**Comment by philerooski** [comment]
I'm seeing this error when I run `devtools::test()`:

```
devtools::test()
Loading mhealthtools
Loading required package: testthat
Testing mhealthtools
✔ | OK F W S | Context
Error in x[[method]](...) : attempt to apply non-function
In addition: Warning message:
`encoding` is deprecated; all files now assumed to be UTF-8 

══ Results ══════════════════════════════════════════════════════════════════════
Duration: 0.2 s

OK:       0
Failed:   2
Warnings: 1
Skipped:  0
```

**Comment by itismeghasyam** [comment]
Yup, it was a sourcing error. I corrected the source path in the test_* files

File: tests/testthat/test_utils.R [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+  # All values are NA except for the last value, our window length is 256, same as the length of gravityVec
+  # min(gravityVec) and max(gravityVec) should be NA, so the testOutput needs to be a dataframe with one obs. for all
+  # the three output vars with max and min having NAs
+  
+  testOutput <- mhealthtools:::tag_outlier_windows_(seq(256), 256, 0.5) # Initialize the data
+  testOutput$max <- NA
+  testOutput$min <- NA
+  
+  expect_equal(mhealthtools:::tag_outlier_windows_(gravityVec, 256, 0.5), testOutput) 
+  # Min and Max should be NA if any window has NAs in it
```
Comment (philerooski) [inline tests/testthat/test_utils.R]: I explicitly inputted na.rm=TRUE when I wrote this function. Is there a good reason we want an entire window min/max to be marked as NA if a single value is NA?

**Comment by itismeghasyam** [comment]
Phil can you check if the new merged repo from your pull request (https://github.com/itismeghasyam/mHealthTools/tree/develop) is consistent (i.e tests run fine, code is not buggy etc.) ?

If so, can you merge this PR into the Sage-Bionetworks develop (https://github.com/Sage-Bionetworks/mhealthtools/tree/develop)

**Comment by philerooski** [comment]
No more test failures.

**Pull request merged**